staging/src/k8s.io/mount-utils/mount_linux_test.go

```diff
@@ -715,19 +699,110 @@ func TestFormat(t *testing.T) {
 			}
 			mounter := NewSafeFormatAndMount(nil, exec, WithMaxConcurrentFormat(tc.max, tc.timeout))
 
-			var wg sync.WaitGroup
+			for i := 0; i < tc.max+1; i++ {
```

Comment:
adds comments on the individual steps so the reader does not have to figure out the logic from teh code

// we run max+1 goroutines and block the commadn execution
// only max goroutine should be running and the additional one should wait
// for one to be releases

something like that ^^^